nifi-nar-bundles/nifi-ignite-bundle/nifi-ignite-processors/src/main/java/org/apache/nifi/processors/ignite/cache/PutIgniteCache.java

```diff
@@ -0,0 +1,373 @@
+    @WritesAttribute(attribute = PutIgniteCache.IGNITE_BATCH_FLOW_FILE_FAILED_COUNT, description = "The total number of failed FlowFiles in the batch"),
+    @WritesAttribute(attribute = PutIgniteCache.IGNITE_BATCH_FLOW_FILE_FAILED_REASON_ATTRIBUTE_KEY, description = "The failed reason attribute key")
+    })
+@ReadsAttributes({
+    @ReadsAttribute(attribute = PutIgniteCache.IGNITE_CACHE_ENTRY_KEY, description = "Ignite cache key"),
+    })
+public class PutIgniteCache extends AbstractIgniteCacheProcessor {
+
+    /**
+     * The batch size of flow files to be processed on invocation of onTrigger
+     */
+    public static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size For Entries")
+            .description("Batch size for entries (1-500).")
+            .defaultValue("250")
+            .required(false)
+            .addValidator(StandardValidators.createLongValidator(1, 500, true))
+            .sensitive(false)
+            .build();
+
+    /**
+     * Data streamer's per node parallelism
+     */
+    public static final PropertyDescriptor DATA_STREAMER_PER_NODE_PARALLEL_OPERATIONS = new PropertyDescriptor.Builder()
+            .name("Data Streamer Per Node Parallel Operations")
+            .description("Data streamer per node parallelism")
+            .defaultValue("5")
+            .required(false)
+            .addValidator(StandardValidators.createLongValidator(1, 10, true))
+            .sensitive(false)
+            .build();
+
+    /**
+     * Data streamers per node buffer size
+     */
+    public static final PropertyDescriptor DATA_STREAMER_PER_NODE_BUFFER_SIZE = new PropertyDescriptor.Builder()
+            .name("Data Streamer Per Node Buffer Size")
+            .description("Data streamer per node buffer size (1-500).")
+            .defaultValue("250")
+            .required(false)
+            .addValidator(StandardValidators.createLongValidator(1, 500, true))
+            .sensitive(false)
+            .build();
+
+    /**
+     * Data streamers auto flush frequency
+     */
+    public static final PropertyDescriptor DATA_STREAMER_AUTO_FLUSH_FREQUENCY = new PropertyDescriptor.Builder()
+            .name("Data Streamer Auto Flush Frequency in millis")
+            .description("Data streamer flush interval in millis")
+            .defaultValue("10")
+            .required(false)
+            .addValidator(StandardValidators.createLongValidator(1, 100, true))
+            .sensitive(false)
+            .build();
+
+    /**
+     * Data streamers override values property
+     */
+    public static final PropertyDescriptor DATA_STREAMER_ALLOW_OVERRIDE = new PropertyDescriptor.Builder()
+            .name("Data Streamer Allow Override")
+            .description("Whether to override values already in the cache")
+            .defaultValue("false")
+            .allowableValues(new AllowableValue("true"), new AllowableValue("false"))
```

Comment:
I believe that when there is a default and allowable values, we want to set .required(true)